packages/extractor/src/internal/Spotify.ts

```diff
@@ -272,6 +272,42 @@ export class SpotifyAPI {
             return null;
         }
     }
+
+    public async getRelatedTracks(trackId: string, artistId: string, limit = 1) {
+        try {
+            // req
+            if (this.isTokenExpired()) await this.requestToken();
+            // failed
+            if (!this.accessToken) return null;
+
+            const res = await fetch(`${SP_BASE}/recommendations/?limit=${limit}&market=US&seed_tracks=${trackId}&seed_artists=${artistId}`, {
```

Comment:
so how do we seed multiple tracks here? joining by `TRACK_ID,TRACK_ID`? What if rest of the tracks in history are not from spotify?

Reply:
Yes, correct. It is a comma separated list of track ids.

We will have to first verify that they are Spotify track IDs before getting used in the API call.

The recommendations API call only allow Spotify IDs, and needs at least one.

Reply:
We have to decide on a way to accept multiple tracks in the Extractor getRelatedTracks() method. Then we can update the internal API accordingly to use multiple tracks.

Reply:
Now takes multiple track ids and artist ids. Spotify recommendations API has a combined max limit of 5, so I also added logic to handle that.